### ISSUE TITLE
training scripts should not evaluate test set

### ISSUE THREAD

**Issue by hhexiy** [dmlc/gluon-nlp#372]
Currently, the MT training scripts do training (with dev set) and evaluation (on the test set) altogether.
Ideally, the training script should not touch test set because reporting numbers on the test set (even if it's not used for model selection) is already leaking test set information.

For example, in [`fairseq`](https://github.com/pytorch/fairseq), train/validation (train.py) and evaluate (generate.py) are in two separate files.

**Comment by avinashsai** [comment]
@szha I would like to take this up. Can you give more context about this?

**Comment by leezu** [comment]
@avinashsai it's about updating the MT training scripts to only evaluate on the dev set to avoid leaking test set information until the users explicitly runs evaluation on the test set

**Comment by avinashsai** [comment]
Fixed via #1228
